beacon-chain/operations/service.go

```diff
@@ -216,18 +217,39 @@ func (s *Service) HandleValidatorExits(ctx context.Context, message proto.Messag
 	return nil
 }
 
-// HandleAttestations processes a received attestation message.
-func (s *Service) HandleAttestations(ctx context.Context, message proto.Message) error {
-	ctx, span := trace.StartSpan(ctx, "operations.HandleAttestations")
+// HandleAttestation processes a received attestation message.
+func (s *Service) HandleAttestation(ctx context.Context, message proto.Message) error {
+	ctx, span := trace.StartSpan(ctx, "operations.HandleAttestation")
 	defer span.End()
 
 	attestation := message.(*ethpb.Attestation)
-	hash, err := hashutil.HashProto(attestation)
+	root, err := ssz.HashTreeRoot(attestation.Data)
 	if err != nil {
 		return err
 	}
-	if s.beaconDB.HasAttestation(hash) {
-		return nil
+	incomingAttBits := attestation.AggregationBits
+	if !bytes.Equal(incomingAttBits.Bytes(), []byte{}) && s.beaconDB.HasAttestation(root) {
+		dbAtt, err := s.beaconDB.Attestation(root)
+		if err != nil {
+			return err
+		}
+
+		if !incomingAttBits.Contains(dbAtt.AggregationBits) {
```

Comment:
I think this is inverted.

Shouldn't it be:
- receive incoming attestation bits
- if this incoming attestation has bits we don't already have in the database,
- aggregate with our existing data
- save the new aggregated attestation